File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/util/Converters.java
package org.kevoree.modeling.api.util;


import java.util.List;

/**
 * Created by duke on 10/3/14.
 */
public class Converters {

    public static String convFlatAtt(Object value) {
        if (value == null) {
            return null;
        }
        if (value instanceof List) {
            boolean isF = true;
            StringBuilder buffer = new StringBuilder();
            for (Object v : (List) value) {
                if (!isF) {
                    buffer.append("$");
                }
                buffer.append(v.toString());
                isF = false;
            }
            return buffer.toString();
        } else {
            return value.toString();
        }
    }

    public static String[] convAttFlat(String value) {
        return value.split("$");
    }

    public static char toChar(byte b) {
        return (char) b;
    }

    public static byte fromChar(char c) {
        return (byte) c;
    }

}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/util/Converters.ts
///<reference path="../../../../../java/util/JUList.ts"/>

class Converters {

  public static convFlatAtt(value: any): string {
    if (value == null) {
      return null;
    }
    if (value instanceof JUList) {
      var isF: boolean = true;
      var buffer: StringBuilder = new StringBuilder();
      //TODO resolve for-each cycle
      var v: any;
      for (v in <JUList>value) {
        if (!isF) {
          buffer.append("$");
        }
        buffer.append(v.toString());
        isF = false;
      }
      return buffer.toString();
    } else {
      return value.toString();
    }
  }

  public static convAttFlat(value: string): string[] {
    return value.split("$");
  }

  public static toChar(b: number): string {
    return <string>b;
  }

  public static fromChar(c: string): number {
    return <number>c;
  }

}

